### PR TITLE
handle unexpected exception on success callback of translog upload

### PR DESCRIPTION
**Pull request by linuxpi**
### Description
- Translog upload success callback can throw unexpected exceptions. 
- Since we are using ActionListener.wrap, we catch all exceptions in onSuccess and call the onFailure callback
- Currently, on failure callback only accepts FileTransferException. 
- if the exception is any other, we throw a ClassCastException and swallow the original exception trace.
- To be able to log the exceptions occurring in onSuccess callbacks, we wrap those exception as FileTransferException 

### Related Issues
https://github.com/opensearch-project/OpenSearch/issues/12544

### Check List
- ~~[ ] New functionality includes testing.~~
  - ~~[ ] All tests pass~~
- ~~[ ] New functionality has been documented.~~
  - ~~[ ] New functionality has javadoc added~~
- [x] Failing checks are inspected and point to the corresponding known issue(s) (See: [Troubleshooting Failing Builds](../blob/main/CONTRIBUTING.md#troubleshooting-failing-builds))
- [x] Commits are signed per the DCO using --signoff
- [x] Commit changes are listed out in CHANGELOG.md file (See: [Changelog](../blob/main/CONTRIBUTING.md#changelog))
- ~~[ ] Public documentation issue/PR [created](https://github.com/opensearch-project/documentation-website/issues/new/choose)~~

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Comment by sachinpkale** [comment]
Changes LGTM. Can we please add UTs for the change?

File: server/src/main/java/org/opensearch/index/translog/transfer/FileTransferTracker.java [hunk 1]
```diff
@@ -64,10 +68,20 @@ long getTotalBytesToUpload() {
...
     @Override
     public void onSuccess(TransferFileSnapshot fileSnapshot) {
-        long durationInMillis = (System.nanoTime() - fileTransferStartTime) / 1_000_000L;
-        remoteTranslogTransferTracker.addUploadTimeInMillis(durationInMillis);
-        remoteTranslogTransferTracker.addUploadBytesSucceeded(bytesForTlogCkpFileToUpload.get(fileSnapshot.getName()));
-        add(fileSnapshot.getName(), TransferState.SUCCESS);
+        try {
+            long durationInMillis = (System.nanoTime() - fileTransferStartTime) / 1_000_000L;
+            remoteTranslogTransferTracker.addUploadTimeInMillis(durationInMillis);
+            remoteTranslogTransferTracker.addUploadBytesSucceeded(bytesForTlogCkpFileToUpload.get(fileSnapshot.getName()));
```
Comment (Bukhtawar) [inline server/src/main/java/org/opensearch/index/translog/transfer/FileTransferTracker.java:74]: Is it that fileSnapshot gets returned as null. Do we need to simply put a null check in that case?

Comment (linuxpi) [reply server/src/main/java/org/opensearch/index/translog/transfer/FileTransferTracker.java:74]: @Bukhtawar fileSnapshot wont be null. The speculation as of now is `bytesForTlogCkpFileToUpload` doesn't contain the entry for `fileSnapshot.getName()`, which leads to null getting assigned to a primitive long, raising NullPointerException. This happens when we get latch timeouts on translog uploads

CC: @sachinpkale 

This try catch we have added is to 1) zero down on this error and 2) to not let any failures to update stats lead to Cx requests failing.

**Comment by kkmr** [comment]
Like @sachinpkale  mentioned, please add unit tests.

**Review by sachinpkale — APPROVED** [review]

**Review by kkmr — APPROVED** [review]

**Pull request merged**